controller/controlplane/controller.go

```diff
@@ -99,7 +100,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	mgrID, err := manager.NewID(string(cp.GetUID()))
+	mgrID, err := manager.NewID(builControlPlaneInstanceID(cp))
```

Comment:
I think that changing this to the proposed nn + uuid makes sense but is there a concrete reason for this? Have you found the current approach limiting in some way?